src/storage.rs

```diff
@@ -353,6 +358,26 @@ impl MemStorage {
         }
     }
 
+    /// Create a new `MemStorage` with a given `Config`. The given `Config` will be used to
+    /// initialize the storage.
+    pub fn new_with_config(cfg: &Config) -> MemStorage {
+        let store = MemStorage::new();
+        store.initialize_with_config(cfg);
+        store
+    }
+
+    /// Initialize a `MemStorage` with a given `Config`.
+    pub fn initialize_with_config(&self, cfg: &Config) {
```

Comment:
Why not use `peers` directly? So that `peers` in `Config` can be removed completely.